src/utils/classifier_report.py

```diff
@@ -0,0 +1,125 @@
+            with a.head():
+                a.meta(charset='utf-8')
+                a.title(_t='Classifier Results')
+                
+                # CSS CDMs
+                a.link(rel="stylesheet", href="https://cdn.datatables.net/1.13.4/css/jquery.dataTables.css")
+                a.link(rel="stylesheet", href="https://cdn.datatables.net/searchpanes/2.1.2/css/searchPanes.dataTables.min.css")
+                a.link(rel="stylesheet", href="https://cdn.datatables.net/select/1.6.2/css/select.dataTables.min.css")
+            
+            # creates table element
+            with a.body():
+                a.table(id="myTable", klass="display")
+                
+                # JQuery Core and UI CDN
+                a.script(src="https://code.jquery.com/jquery-3.7.0.js", integrity="sha256-JlqSTELeR4TLqP0OG9dxM7yDPqX1ox/HfgiSLBj8+kM=", crossorigin="anonymous")
+                a.script(src="https://code.jquery.com/ui/1.13.2/jquery-ui.js", integrity="sha256-xLD7nhI62fcsEZK2/v8LsBcb4lG7dgULkuXoXB/j91c=", crossorigin="anonymous")
+
+                # Datables CDN
+                a.script(src="https://cdn.datatables.net/1.13.4/js/jquery.dataTables.js")
+                a.script(src="https://cdn.datatables.net/searchpanes/2.1.2/js/dataTables.searchPanes.min.js")
+                a.script(src="https://cdn.datatables.net/select/1.6.2/js/dataTables.select.min.js")
+                a.script(src="https://cdn.datatables.net/fixedheader/3.3.2/js/dataTables.fixedHeader.min.js")
+                a.script(src="https://cdn.datatables.net/responsive/2.4.1/js/dataTables.responsive.min.js")
+                
+                # Javascript creating the table
+                #  the 'Path' column accounts for the csv 
+                with a.script():
+                    a(f'''
+                        let table = new DataTable('#myTable', {{
+                            data: {self.data},
+                            responsive: true,
+                            fixedHeader:{{ header: true, footer: true }},
+                            ordering: true,
+                            paging: true,
+                            scrollY: true,
+                            searching: true,
+                            searchPanes: {{viewTotal: true, layout: 'columns-4', initCollapsed: true}},
+                            dom: 'Plfrtip',
+                            columns: [
+                                {{ title: 'Image', data: 'normalized_path', render: (d,t,r,m) => '<img src="'+d+'" style=height:200px; />'}},
+                                {{ title: 'Path', data: 'original_path'}},
+                                {{ title: 'Class', data: 'predicted_class'}},
+                                {{ title: 'Confidence', data: 'predicted_conf'}}
+                            ],
+                        }});
+                    ''')
+        
+        self.html_page = a
+    
+    # launches the HTML page in default browser
+    def launch_page(self):
```

Comment:
I would suggest splitting this into two methods: 1) serializing the html report to a file, 2) launching the report in a browser. They seem like separate operations, and it seems like a user might want to save the report without opening it.